### PR TITLE
Fix AI superweapon usage.

### PR DESCRIPTION
**Pull request by GraionDilach**
For testcase, just spectate an AI match on http://resource.openra.net/maps/17498.

### CODE REVIEW

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 1]
```diff
@@ -1027,7 +1027,8 @@ void TryToUseSupportPower(Actor self)
 					// Valid target found, delay by a few ticks to avoid rescanning before power fires via order
 					BotDebug("AI: {2} found new target location {0} for support power {1}.", attackLocation, sp.Info.OrderName, Player.PlayerName);
 					waitingPowers[sp] += 10;
-					QueueOrder(new Order(sp.Info.OrderName, supportPowerMngr.Self, false) { TargetLocation = attackLocation.Value, SuppressVisualFeedback = true });
+					var activateOrderName = sp.Info.AllowMultiple ? sp.Info.OrderName + "_" + sp.Instances.First().Self.ActorID : sp.Info.OrderName;
```
Comment (pchote) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: Surely this should be selecting the first *active* instance?

Comment (GraionDilach) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: If AllowMultiple is enabled then each support power will only have a single instance in the list.

**Comment by GraionDilach** [comment]
Updated, now SupportPowerInstance.key is exposed for this.

File: OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs [hunk 2]
```diff
@@ -170,7 +171,7 @@ public class SupportPowerInstance
 		public SupportPowerInstance(string key, SupportPowerManager manager)
 		{
 			this.manager = manager;
-			this.key = key;
+			this.Key = key;
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs]: `this` is no longer necessary.

**Comment by GraionDilach** [comment]
Updated again.

Also added a second commit which fixes AI support power targeting on `RectangularIsometric` maps.

**Review by pchote — APPROVED** [review]
Code changes look fine, and tested that the AI can still use powers in TD.

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/63cc497e0d60e31e131f4d76afd64ccf0fe851dd...b34a3b82071f9d588be6f75428fa9df5a3dd9304)

**Pull request merged**